src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -148,7 +157,7 @@ private static RepoConfiguration getMatchingRepoConfig(
             List<RepoConfiguration> repoConfigs, AuthorConfiguration authorConfig) {
         for (RepoConfiguration repoConfig : repoConfigs) {
             if (repoConfig.getLocation().equals(authorConfig.getLocation())
-                    && repoConfig.getBranch().equals(authorConfig.getBranch())) {
+                    && (authorConfig.isDefaultBranch() || repoConfig.getBranch().equals(authorConfig.getBranch()))) {
```

Comment:
Can use a descriptive boolean for this condition, to improve readability?
Also, you can swap `authorConfig` and `repoConfig` around in the second part of the condition.

i.e. `boolean isDefaultBranchOrRepoConfigBranch = (authorConfig.isDefaultBranch() || authorConfig.getBranch().equals(repoConfig.getBranch()))`